### PR TITLE
*: context.Context instead of map[interface{}]interface{}

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: expression/builtin/builtin.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ package builtin
 
 import (
 	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
 )
 
 const (
```
Comment (shenli) [inline expression/builtin/builtin.go]: ExprEvalArgCtx is useless now.

File: expression/builtin/info.go [hunk 2]
```diff
@@ -40,44 +35,44 @@ func builtinDatabase(args []interface{}, data map[interface{}]interface{}) (v in
...
 
 // See: https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_current-user
 // TODO: The value of CURRENT_USER() can differ from the value of USER(). We will finish this after we support grant tables.
-func builtinCurrentUser(args []interface{}, data map[interface{}]interface{}) (v interface{}, err error) {
-	c, ok := data[ExprEvalArgCtx]
-	if !ok {
+func builtinCurrentUser(args []interface{}, ctx context.Context) (v interface{}, err error) {
+	data := variable.GetSessionVars(ctx)
+	if data == nil {
 		return nil, errors.Errorf("Missing ExprEvalArgCtx when evalue builtin")
```
Comment (shenli) [inline expression/builtin/info.go]: This will never happened.

File: expression/builtin/info.go [hunk 3]
```diff
@@ -40,44 +35,44 @@ func builtinDatabase(args []interface{}, data map[interface{}]interface{}) (v in
...
 }
 
 // See: https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_current-user
 // TODO: The value of CURRENT_USER() can differ from the value of USER(). We will finish this after we support grant tables.
-func builtinCurrentUser(args []interface{}, data map[interface{}]interface{}) (v interface{}, err error) {
-	c, ok := data[ExprEvalArgCtx]
-	if !ok {
+func builtinCurrentUser(args []interface{}, ctx context.Context) (v interface{}, err error) {
+	data := variable.GetSessionVars(ctx)
+	if data == nil {
```
Comment (coocood) [inline expression/builtin/info.go]: check `ctx == nil` then `GetSessionVars(ctx)`

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**